presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/Util.java

```diff
@@ -115,4 +120,72 @@ public static PlanNode transpose(PlanNode parent, PlanNode child)
         }
         return Optional.of(node.replaceChildren(newChildrenBuilder.build()));
     }
+
+    /**
+     * Looks for the pattern (ProjectNode*)TargetNode, and rewrites it to TargetNode(ProjectNode*),
+     * returning an empty option if the pattern doesn't match.
```

Comment:
Is it always possible to reorder such nodes? What if project nodes uses output of target node?